[ui] Hide __ASSET_JOB references on the sensor pages

bengotow opened the pull request:
## Summary & Motivation

This is a quick fix for the issue raised in https://github.com/dagster-io/dagster/issues/14844. We don't have the information about the exact assets in your selection, but we can see that the targets are asset jobs and show "A selection of assets". (Fixes https://github.com/dagster-io/dagster/issues/14844)

I think there are a few other things we should consider changing:

- Backfills have "targets" and sensors have "Asset / Job". I'm tempted to change the labeling in the table + details page to say "Target" to match the backfill terminology.

- We are currently showing `sensor.metadata.assetKeys` in the Asset / Job column. Confusingly these are not the targeted asset keys, they're the monitored asset keys (if the sensor is an @asset_sensor). To make this more clear in the code I wrapped the rendering into a component called `<SensorMonitoredAssets />`, but the placement of them in the table (beneath the "A selection of assets" or job target info) might be confusing to users. I don't have a better placement idea - perhaps we just remove them from the list view. 

<img width="1400" alt="image" src="https://github.com/dagster-io/dagster/assets/1037212/4c5f2409-62e0-42ab-9ed2-4b1a79f746a0">

Loom of this: https://www.loom.com/share/056baeb022fe4c95874656b2a576267b

## How I Tested These Changes

I tested these with a few asset_sensor, op job and asset job sensors

bengotow commented on js_modules/dagit/packages/core/src/sensors/SensorDetails.tsx:
```diff
@@ -70,30 +67,8 @@ export const SensorDetails: React.FC<{
-      return targetCount > 1 ? 'Jobs / Pipelines' : 'Pipeline';
-    }
-    return targetCount > 1 ? 'Jobs' : 'Job';
-  }, [anyPipelines, targetCount]);
```
All of this code was determining whether to show Jobs / Pipelines, Pipelines or Job. In the list view we just say `Asset / Job` and I think doing that in the detail view is better for consistency + much simpler 😅

sryza commented:
+1 to these changes. And I like "Target" over "Asset / Job".

> We are currently showing sensor.metadata.assetKeys in the Asset / Job column. Confusingly these are not the targeted asset keys, they're the monitored asset keys (if the sensor is an @asset_sensor). To make this more clear in the code I wrapped the rendering into a component called <SensorMonitoredAssets />, but the placement of them in the table (beneath the "A selection of assets" or job target info) might be confusing to users. I don't have a better placement idea - perhaps we just remove them from the list view.

I agree that this is confusing. I think we should take that information out of the Asset / Job column, even if it means we just don't show it in the table view at all.

bengotow commented:
Thanks for the feedback @sryza! I've updated this to use Target instead of Asset / Job and removed the "Monitored Assets" display from the list view. I think it was definitely confusing in the target column and doesn't seem worth adding another column and crunching down all the existing data
